LEAF_Nexus/sources/ActiveDirectory.php

```diff
@@ -29,23 +29,53 @@ public function searchMember(string $uid = "") : ?array
       $info = [];
       $filter = "(&(objectCategory=Person)(objectClass=User)(anr=$uid))"; // search only for users classified as persons (i.e. not service accounts)
       $attr = ["objectClass", "sn", "givenName","initials","title","description","telephoneNumber","mail","sAMAccountName","objectGUID","mobile","physicalDeliveryOfficeName"]; // fields for each result to return
-      if ($this->bind) {
-         $results = ldap_search($this->conn, $this->base, $filter, $attr);
-         $info = ldap_get_entries($this->conn, $results);
+      if ($this->bind === false) {
+         $return_value = array(
+            "status" => array(
+               "code" => 4,
+               "message" => "The connection to the LDAP server failed."
+            )
+         );
+         return $return_value;
       }
-      return $info;
+

```

Comment:
It's better to have an else here and only one return for this method. If you have an else here you won't need the return on line 39 you can wait and only have the one return on line 51